### PR TITLE
Add remaining missing return types to safe methods

### PR DESCRIPTION
**Pull request by wouterj**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Ref #47551
| License       | MIT
| Doc PR        | -

This adds return types to remaining methods that are safe to receive a return type (final, internal or private ones), except from methods of `DataCollector` classes (which are missing quite a lot of types). I couldn't figure out whether they all return `Data`, the actual value or a union of both, so I've left them out for now.

Asides from these, we're missing 121 more return types on "non-safe" methods. If I have time, I'll create some smaller PRs for them to allow easier reviews.

<details>
<summary>Full list</summary>

```120 missing return types on interfaces

Symfony\Bridge\Doctrine\DependencyInjection\CompilerPass\RegisterEventListenersAndSubscribersPass::process
Symfony\Bridge\Doctrine\Form\DoctrineOrmTypeGuesser::getMetadata
Symfony\Bridge\Monolog\Handler\ConsoleHandler::setOutput
Symfony\Bridge\Monolog\Handler\ConsoleHandler::onCommand
Symfony\Bridge\Monolog\Handler\ConsoleHandler::onTerminate
Symfony\Bridge\Monolog\Handler\ServerLogHandler::createSocket
Symfony\Bundle\FrameworkBundle\Command\AbstractConfigCommand::listBundles
Symfony\Bundle\FrameworkBundle\Command\AbstractConfigCommand::validateConfiguration
Symfony\Bundle\FrameworkBundle\DependencyInjection\Configuration::addHttpClientRetrySection
Symfony\Bundle\FrameworkBundle\DependencyInjection\Configuration::addRemoteEventSection
Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension::registerWebhookConfiguration
Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension::registerRemoteEventConfiguration
Symfony\Bundle\SecurityBundle\DependencyInjection\SecurityExtension::createContextListener
Symfony\Component\BrowserKit\AbstractBrowser::getScript
Symfony\Component\Cache\Adapter\ArrayAdapter::freeze
Symfony\Component\Cache\Adapter\ArrayAdapter::unfreeze
Symfony\Component\Cache\Adapter\MemcachedAdapter::checkResultCode
Symfony\Component\Cache\DependencyInjection\CachePoolPass::process
Symfony\Component\Config\Definition\BaseNode::validateType
Symfony\Component\Config\Definition\PrototypedArrayNode::addChild
Symfony\Component\Config\FileLocator::locate
Symfony\Component\Config\Loader\FileLoader::setCurrentDir
Symfony\Component\Config\Loader\FileLoader::import
Symfony\Component\Config\Loader\FileLoader::doImport
Symfony\Component\Config\ResourceCheckerConfigCache::safelyUnserialize
Symfony\Component\Console\Command\LockableTrait::release
Symfony\Component\Console\Formatter\OutputFormatter::formatAndWrap
Symfony\Component\Console\Helper\Helper::formatTime
Symfony\Component\Console\Helper\Table::setStyleDefinition
Symfony\Component\Console\Helper\Table::render
Symfony\Component\Console\Helper\Table::renderRowSeparator
Symfony\Component\Console\Helper\Table::renderRow
Symfony\Component\Console\Helper\Table::calculateNumberOfColumns
Symfony\Component\Console\Helper\Table::calculateColumnsWidth
Symfony\Component\Console\Helper\Table::cleanup
Symfony\Component\Console\Input\Input::parse
Symfony\Component\Console\Input\Input::getStream
Symfony\Component\Console\Output\Output::doWrite
Symfony\Component\DependencyInjection\Argument\ReferenceSetArgumentTrait::setValues
Symfony\Component\DependencyInjection\Container::make
Symfony\Component\DependencyInjection\Container::load
Symfony\Component\DependencyInjection\Extension\Extension::getXsdValidationBasePath
Symfony\Component\DependencyInjection\Extension\Extension::getNamespace
Symfony\Component\DependencyInjection\Extension\Extension::getConfiguration
Symfony\Component\DependencyInjection\Loader\YamlFileLoader::parseDefinition
Symfony\Component\DependencyInjection\ParameterBag\FrozenParameterBag::clear
Symfony\Component\DependencyInjection\ParameterBag\FrozenParameterBag::add
Symfony\Component\DependencyInjection\ParameterBag\FrozenParameterBag::set
Symfony\Component\DependencyInjection\ParameterBag\FrozenParameterBag::deprecate
Symfony\Component\DependencyInjection\ParameterBag\FrozenParameterBag::remove
Symfony\Component\DependencyInjection\ParameterBag\ParameterBag::isResolved
Symfony\Component\DomCrawler\AbstractUriElement::setNode
Symfony\Component\DomCrawler\Field\FormField::initialize
Symfony\Component\ExpressionLanguage\Compiler::getFunction
Symfony\Component\ExpressionLanguage\Node\Node::toArray
Symfony\Component\ExpressionLanguage\Parser::parseExpression
Symfony\Component\ExpressionLanguage\Parser::getPrimary
Symfony\Component\ExpressionLanguage\SerializedParsedExpression::getNodes
Symfony\Component\Filesystem\Filesystem::linkException
Symfony\Component\Form\AbstractType::getBlockPrefix
Symfony\Component\Form\AbstractType::getParent
Symfony\Component\Form\ButtonBuilder::setFormFactory
Symfony\Component\Form\Extension\Core\DataAccessor\PropertyPathAccessor::getPropertyValue
Symfony\Component\Form\Extension\Core\Type\ChoiceType::buildForm
Symfony\Component\Form\Extension\Core\Type\ChoiceType::buildView
Symfony\Component\Form\Extension\Core\Type\ChoiceType::finishView
Symfony\Component\Form\Extension\Core\Type\ChoiceType::configureOptions
Symfony\Component\Form\Extension\Core\Type\ChoiceType::addSubForms
Symfony\Component\Form\Extension\Core\Type\ChoiceType::addSubForm
Symfony\Component\Form\Extension\Core\Type\DateType::buildForm
Symfony\Component\Form\Extension\Core\Type\DateType::finishView
Symfony\Component\Form\Extension\Core\Type\DateType::configureOptions
Symfony\Component\Form\Extension\Core\Type\FileType::buildForm
Symfony\Component\Form\Extension\Core\Type\FileType::buildView
Symfony\Component\Form\Extension\Core\Type\FileType::finishView
Symfony\Component\Form\Extension\Core\Type\FileType::configureOptions
Symfony\Component\Form\Extension\Core\Type\MoneyType::getPattern
Symfony\Component\Form\Extension\Validator\ViolationMapper\ViolationPathIterator::mapsForm
Symfony\Component\HttpClient\Response\StreamWrapper::stream_cast
Symfony\Component\HttpFoundation\Request::prepareRequestUri
Symfony\Component\HttpKernel\DependencyInjection\ConfigurableExtension::loadInternal
Symfony\Component\Ldap\Security\LdapUserProvider::getAttributeValue
Symfony\Component\Mailer\Bridge\Mailjet\Transport\MailjetApiTransport::castCustomHeader
Symfony\Component\Mime\Email::ensureValidity
Symfony\Component\Process\Exception\ProcessFailedException::getProcess
Symfony\Component\Process\Exception\ProcessTimedOutException::getProcess
Symfony\Component\Process\Exception\ProcessTimedOutException::getExceededTimeout
Symfony\Component\Process\Process::start
Symfony\Component\Process\Process::checkTimeout
Symfony\Component\Process\Process::setOptions
Symfony\Component\Process\Process::updateStatus
Symfony\Component\Process\Process::readPipesForOutput
Symfony\Component\Process\Process::readPipes
Symfony\Component\Process\Process::resetProcessData
Symfony\Component\Process\Process::requireProcessIsStarted
Symfony\Component\Process\Process::requireProcessIsTerminated
Symfony\Component\Routing\Loader\AnnotationClassLoader::configureRoute
Symfony\Component\Routing\Matcher\UrlMatcher::getExpressionLanguage
Symfony\Component\Routing\Router::getRouteCollection
Symfony\Component\Security\Core\User\InMemoryUserProvider::createUser
Symfony\Component\Security\Http\Firewall::getSubscribedEvents
Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer::supportsNormalization
Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer::normalize
Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer::instantiateObject
Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer::supportsDenormalization
Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer::denormalize
Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer::setAttributeValue
Symfony\Component\Translation\Catalogue\AbstractOperation::processDomain
Symfony\Component\Translation\Command\XliffLintCommand::display
Symfony\Component\Translation\Command\XliffLintCommand::displayJson
Symfony\Component\Translation\Command\XliffLintCommand::getDirectoryIterator
Symfony\Component\Translation\Command\XliffLintCommand::isReadable
Symfony\Component\Translation\Extractor\PhpExtractor::seekToNextRelevantToken
Symfony\Component\Translation\Extractor\PhpExtractor::skipMethodArgument
Symfony\Component\Translation\Extractor\PhpExtractor::parseTokens
Symfony\Component\Translation\Util\ArrayConverter::getElementByPath
Symfony\Component\Validator\Constraints\IsbnValidator::getMessage
Symfony\Component\Validator\Exception\ValidationFailedException::getValue
Symfony\Component\VarDumper\Test\VarDumperTestTrait::assertDumpEquals
Symfony\Component\VarDumper\Test\VarDumperTestTrait::assertDumpMatchesFormat
```

</details>

### CODE REVIEW

File: src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php [hunk 1]
```diff
@@ -379,7 +381,7 @@ private function createQueryBuilder(string $alias = 'm'): QueryBuilder
         ));
     }
 
-    private function executeQuery(string $sql, array $parameters = [], array $types = [])
+    private function executeQuery(string $sql, array $parameters = [], array $types = []): Result|AbstractionResult|Statement|DriverResult
```
Comment (wouterj) [inline src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php]: This is what I derived from the mocks used in the test suite, but it looks way too complex given we only support 2 major DBAL versions (so I would expect at most an union of 2 classes).

Does someone know the actual return type here, so we can update the mocks in the test suite?

Comment (nicolas-grekas) [reply src/Symfony/Component/Messenger/Bridge/Doctrine/Transport/Connection.php]: I dug a bit the git history of DBAL, it looks like `Result|AbstractionResult|ResultStatement` could work.
I pushed it, let's see.

**Review by nicolas-grekas — APPROVED** [review]
As always, mocks are desynchronized from the real implementation and make use move forward with closed eyes :(
I wish we could stop using them as much as possible, and use e.g. an sqlite DB here.

**Comment by nicolas-grekas** [comment]
Thank you @wouterj.

File: src/Symfony/Bridge/Monolog/Handler/NotifierHandler.php [hunk 2]
```diff
@@ -71,7 +71,7 @@ private function notify(array $records): void
         $this->notifier->send($notification, ...$this->notifier->getAdminRecipients());
     }
 
-    private function getHighestRecord(array $records)
+    private function getHighestRecord(array $records): array
```
Comment (pauliuspetronis) [inline src/Symfony/Bridge/Monolog/Handler/NotifierHandler.php:74]: I think return must be `: LogRecord|array` for  monolog 3+ compatibility

**Pull request merged**